packages/@core.keystone/errors.js

```diff
@@ -0,0 +1,128 @@
+ *     UNABLE_EXECUTE_SOME_PROCEDURE: { mutation: 'myCustomSchema', code: GQLErrorCode.INTERNAL_ERROR, message: `Oops, something went wrong` },
+ * }
+ *
+ * const MyCustomSchema = new GQLCustomSchema('myCustomAction', {
+ *     types: [
+ *         {
+ *             access: true,
+ *             type: 'input MyCustomActionInput { dv: Int!, sender: SenderFieldInput!, phone: String, userId: String }',
+ *         }
+ *     ],
+ *     mutations: [
+ *         {
+ *             access: true,
+ *             schema: 'myCustomAction(data: MyCustomActionInput!): User',
+ *             doc: {
+ *                 summary: 'Does this and that, shortly',
+ *                 description: 'More detailed explanation goes here',
+ *                 // Pass declared errors object here to get formatted documentation in GraphiQL
+ *                 errors,
+ *             },
+ *             resolver: async (parent, args, context) => {
+ *                 // validate passed arguments
+ *                 if (!normalizePhone(args.phone)) {
+ *                     // Use our customized error class and pass appropriate error into it
+ *                     throw new GQLError(errors.WRONG_PHONE_FORMAT)
+ *                 }
+ *
+ *                 // find some record
+ *                 const user = await User.getOne(context, { id: args.userId })
+ *                 if (!user) {
+ *                     throw new GQLError(errors.UNABLE_TO_FIND_USER)
+ *                 }
+ *
+ *                 // execute some another actions, that is not guaranteed to be succeed
+ *                 try {
+ *                     executeSomeAnotherAction
+ *                 } catch(e) {
+ *                     throw new GQLError({
+ *                         ...errors.UNABLE_EXECUTE_SOME_PROCEDURE,
+ *                         internalError: e
+ *                     })
+ *                 }
+ *             }
+ *         }
+ *     ]
+ * })
+ * ```
+ */
+const { ApolloError } = require('apollo-server-errors')
+
+// Unable to find a record, whose identifier is specified in some argument of query or mutation
+const NOT_FOUND = 'NOT_FOUND'
+// Data, being submitted, conflicts with constraints of a data storage
+// For example, it can violate constraints on uniqueness, number of records, formats of field values, values itself etc.
+const CONFLICT = 'CONFLICT'
+// Wrong format or not enough data in user input
+const BAD_USER_INPUT = 'BAD_USER_INPUT'
+// Generic error, that something went wrong at server side, though user input was correct
+const INTERNAL_ERROR = 'INTERNAL_ERROR'
+// Access denied. Semantically it corresponds to 403 HTTP code
+const FORBIDDEN = 'FORBIDDEN'
+
+
+/**
+ * First level of error classification, used in custom GraphQL queries or mutations
+ * Only generic error kinds are listed
+ * Conceptually, it conforms to HTTP standard for error codes
+ * @readonly
+ * @enum {String}
+ */
+const GQLErrorCode = {
+    NOT_FOUND,
+    CONFLICT,
```

Comment:
https://www.apollographql.com/docs/apollo-server/data/errors/#error-codes 
There is no conflict here. It's better to use simplified APOLLO codes. It's really not the same as HTTP codes but similar to them

Reply:
I think the CONFLICT is the same as BAD_USER_INPUT. I don't know any useful case for the code. Do you know any?

Reply:
No, they are different.

`BAD_USER_INPUT` is about input data, that does not matches required format.

`CONFLICT` is about state of a storage, that violates existing constraints
Take a look at a [W3C Standard](https://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html)
> The request could not be completed due to a conflict with the current state of the resource

Reply:
> https://www.apollographql.com/docs/apollo-server/data/errors/#error-codes
> There is no CONFLICT here.
Apollo errors are about handling some abstract GraphQL, so, we see a limited set of errors.
I'm developing errors standard, that is more about business logic and Apollo errors don't fit for that.